### PR TITLE
Add optional claim-backed variants of assisted-service jobs

### PR DESCRIPTION
**Pull request by petr-muller**
/cc @ghost

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 1]
```diff
@@ -81,15 +81,40 @@ tests:
...
+    achitecture: amd64
+    cloud: aws
+    owner: dpp
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.8"
   steps:
-    cluster_profile: aws
+    pre:
+    - as: expose-ci-images
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:93]: Why not use the `ref` ?

Comment (petr-muller) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:93]: b/c the ref does not work when KUBECONFIG is set to cluster under test (=always in jobs running on claims). That's why I unset KUBECONFIG few lines below. I think I can just add the unset to the ref and use that, but I wanted to test if this alone will be enough to make the job work.

Comment (stevekuznetsov) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:93]: aaah!

**Comment by petr-muller** [comment]
I want to get this tested, so moving from draft state. The definitions are very verbose now because I inlined the common steps for testing. After https://github.com/openshift/release/pull/19337 merges, they can be removed and the jobs can simply use the workflow added there.

/hold

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**